src/funcx_common/task_storage/s3.py

```diff
@@ -11,55 +11,100 @@
 from .base import StorageException, TaskStorage
 
 
-class S3TaskStorage(TaskStorage):
+class RedisS3Storage(TaskStorage):
     """
-    Store Task data in S3
-
+    Abstract storage over Redis and S3.
+    Uses Redis to store objects below of size threshold, and S3 for the rest
     """
 
-    storage_id = "S3TaskStorage"
-
-    def __init__(self, bucket_name: str) -> None:
+    def __init__(self, bucket_name: str, redis_threshold: int = 20000) -> None:
+        """
+        :param bucket_name: Name of the S3 bucket to use
+        :param redis_threshold: Max size of the data that redis accommodates
```

Comment:
I don't need `chars` in the name, but can we include in the docstring here that size is measured in characters, not bytes? Sooner or later, we'll have some 2-byte and 4-byte characters in a result and we could get confused if this isn't recorded.